src/libraries/System.Security.Cryptography.Cose/tests/CoseMessageTests.DecoseSign1.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Formats.Cbor;
+using Test.Cryptography;
+using Xunit;
+using static System.Security.Cryptography.Cose.Tests.CoseTestHelpers;
+
+namespace System.Security.Cryptography.Cose.Tests
+{
+    public class CoseMessageTests
+    {
+        [Fact]
+        public void DecodeVerifyUntagged()
+        {
+            CoseSign1Message msg = CoseMessage.DecodeSign1(ByteUtils.HexToByteArray("8445A201260300A10442313154546869732069732074686520636F6E74656E742E58406520BBAF2081D7E0ED0F95F76EB0733D667005F7467CEC4B87B9381A6BA1EDE8E00DF29F32A37230F39A842A54821FDD223092819D7728EFB9D3A0080B75380B"));
+            Assert.True(msg.Verify(ECDsaKeys[ECDsaAlgorithm.ES256]));
+        }
+
+        [Fact]
+        public void DecodeVerifyDetachedContent()
+        {
+            CoseSign1Message msg = CoseMessage.DecodeSign1(ByteUtils.HexToByteArray("D28445A201260300A104423131F658406520BBAF2081D7E0ED0F95F76EB0733D667005F7467CEC4B87B9381A6BA1EDE8E00DF29F32A37230F39A842A54821FDD223092819D7728EFB9D3A0080B75380B"));
+            Assert.Null(msg.Content);
+            Assert.True(msg.Verify(ECDsaKeys[ECDsaAlgorithm.ES256], s_SampleContent));
+        }
+
+        [Theory]
+        // Protected bucket has duplicate header.
+        [InlineData(true, "D28446A20126013822A10442313154546869732069732074686520636F6E74656E742E584087DB0D2E5571843B78AC33ECB2830DF7B6E0A4D5B7376DE336B23C591C90C425317E56127FBE04370097CE347087B233BF722B64072BEB4486BDA4031D27244F")]
+        // Unprotected bucket has duplicate header.
+        [InlineData(true, "D28441A0A301260138220442313154546869732069732074686520636F6E74656E742E584087DB0D2E5571843B78AC33ECB2830DF7B6E0A4D5B7376DE336B23C591C90C425317E56127FBE04370097CE347087B233BF722B64072BEB4486BDA4031D27244F")]
+        // Duplicate header is in the union of protected and unprotected buckets.
+        [InlineData(false, "D28443A10126A201260442313154546869732069732074686520636F6E74656E742E584087DB0D2E5571843B78AC33ECB2830DF7B6E0A4D5B7376DE336B23C591C90C425317E56127FBE04370097CE347087B233BF722B64072BEB4486BDA4031D27244F")]
+        public void DecodeThrowsWithDuplicateHeaders(bool shouldContainInnerException, string hexCborMessage)
```

Comment:
The feedback about "check how the test case is identified" applies to all `[Theory]`s.  I'm pretty sure this one will come out pretty weak.

I'd make this private, and then make three `[Fact]`s that feed into it (e.g. DuplicateProtectedHeaderThrows, DuplicateUnprotectedHeaderThrows, ProtectedAndUnprotectedHeaderThrows)